casper/src/test/scala/coop/rchain/casper/api/ListeningNameAPITest.scala

```diff
@@ -0,0 +1,145 @@
+package coop.rchain.casper.api
+
+import cats.{ApplicativeError, Id}
+import cats.data.EitherT
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.catscontrib.TaskContrib.TaskOps
+import coop.rchain.casper.genesis.Genesis
+import coop.rchain.casper.helper.{
+  BlockStoreFixture,
+  BlockStoreTestFixture,
+  CasperEffect,
+  HashSetCasperTestNode
+}
+import coop.rchain.casper.protocol._
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.casper.util.rholang.InterpreterUtil
+import coop.rchain.casper.util.rholang.RuntimeManager
+import coop.rchain.comm.transport
+import coop.rchain.comm.transport.CommMessages.packet
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.crypto.signatures.Ed25519
+import coop.rchain.models.{Channel, Expr, PCost, Par}
+import coop.rchain.rholang.interpreter.Runtime
+import java.nio.file.Files
+
+import coop.rchain.blockstorage.BlockStore
+import coop.rchain.casper.HashSetCasperTest.createGenesis
+import coop.rchain.casper.{
+  HashSetCasperTest,
+  MultiParentCasper,
+  MultiParentCasperConstructor,
+  SafetyOracle
+}
+import coop.rchain.casper.genesis.contracts.ProofOfStakeValidator
+import coop.rchain.models.Channel.ChannelInstance.Quote
+import coop.rchain.models.Expr.ExprInstance.{GInt, GString}
+import coop.rchain.p2p.EffectsTestInstances.LogStub
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Matchers}
+import coop.rchain.shared.PathOps.RichPath
+
+import scala.collection.immutable
+
+class ListeningNameAPITest extends FlatSpec with Matchers with BlockStoreFixture {
+
+  import HashSetCasperTest._
+
+  private val (validatorKeys, validators) = (1 to 4).map(_ => Ed25519.newKeyPair).unzip
+  private val genesis                     = createGenesis(validators)
+
+  "getListeningNameResponse" should "return listening names across a chain" in {
+    val stake                 = 10
+    val equalBonds            = validators.zipWithIndex.map { case (v, _) => v -> stake }.toMap
+    val genesisWithEqualBonds = buildGenesis(equalBonds)
+    val nodes                 = HashSetCasperTestNode.network(validatorKeys.take(3), genesisWithEqualBonds)
+    implicit val nodeZeroConstructor = MultiParentCasperConstructor
```

Comment:
The node must already have an instance of this. Why create another one instead of accessing the effect like you do for the safety oracle and logs?